src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1190,12 +1230,17 @@ private void VisitObjectOrDynamicObjectCreation(BoundExpression node, BoundExpre
                 bool isTrackableStructType = EmptyStructTypeCache.IsTrackableStructType(type);
                 if (!type.IsValueType || isTrackableStructType)
                 {
-                    receiver = GetOrCreateObjectCreationPlaceholder(node);
-                    slot = GetOrCreateSlot(receiver);
+                    slot = GetOrCreateObjectCreationPlaceholderSlot(node);
                     if (slot > 0 && isTrackableStructType)
                     {
                         this.State[slot] = NullableAnnotation.NotNullable;
-                        InheritNullableStateOfTrackableStruct(type, slot, valueSlot: -1, isByRefTarget: false, slotWatermark: GetSlotWatermark());
+                        InheritNullableStateOfTrackableStruct(
+                            type,
+                            slot,
+                            valueSlot: -1,
+                            isDefaultValue: (node as BoundObjectCreationExpression)?.Constructor.IsImplicitConstructor == true,
```

Comment:
>(node as BoundObjectCreationExpression)?.Constructor.IsImplicitConstructor == true [](start = 44, length = 82)

This doesn't look like a test for a Default Value Type Constructor. Am I misinterpreting the intent?

Reply:
Changed.

---
In reply to: [246953718](https://github.com/dotnet/roslyn/pull/32304#discussion_r246953718) [](ancestors = 246953718)